lib/internal/fs/streams.js

```
@@ -178,7 +182,12 @@ ReadStream.prototype._read = function(n) {
     return this.push(null);
 
   // the actual read.
+  this.isPerformingIO = true;
   fs.read(this.fd, pool, pool.used, toRead, this.pos, (er, bytesRead) => {
+    this.isPerformingIO = false;
+    // Tell ._destroy() that it's safe to close the fd now.
+    if (this.destroyed) return this.emit(kIoDone);
```

This will swallow the error.

---

```js
if (this.destroyed) {
  this.emit(kIoDone, er); // Don't swallow error
  return
}
```